### PR TITLE
Report Template additional operations

### PR DESCRIPTION
**Pull request by lhellebr**

### CODE REVIEW

**Comment by lhellebr** [comment]
Used by https://github.com/SatelliteQE/robottelo/pull/7437

**Comment by JacobCallahan** [comment]
@lhellebr can you elaborate on this PR? There is no commit message.

**Comment by lhellebr** [comment]
This PR implements some API operations for Reporting feature as needed by linked PR.

**Comment by lhellebr** [comment]
https://github.com/SatelliteQE/robottelo/pull/7437 which **depends** on this PR has been merged already so this needs to get merged quick. @JacobCallahan , can you please take a look?

**Comment by ntkathole** [comment]
ok to test : tests/foreman/api/test_reporttemplates.py

File: nailgun/entities.py [hunk 1]
```diff
@@ -1892,6 +1893,22 @@ def clone(self, synchronous=True, **kwargs):
         response = client.post(self.path('clone'), **kwargs)
         return _handle_response(response, self._server_config, synchronous)
 
+    def generate(self, synchronous=True, **kwargs):
```
Comment (ntkathole) [inline nailgun/entities.py]: Add coverage for this method https://github.com/SatelliteQE/nailgun/blob/master/tests/test_entities.py#L338

**Comment by lhellebr** [comment]
Test added

**Comment by mirekdlugosz** [comment]
@lhellebr Looks like your test failed during travis run on Python 3. Curiously enough, it passed on Python 2. I think you are missing `config.ServerConfig()` call.

**Comment by lhellebr** [comment]
Interesting. Fixed.

**Comment by JacobCallahan** [comment]
I'm a T1 reviewer in this repo, so you can't merge on my ACK. But this looks good to me, thanks for the update!

File: nailgun/entities.py [hunk 2]
```diff
@@ -1868,8 +1869,8 @@ def path(self, which=None):
         ``super`` is called otherwise.
 
         """
-        if which == 'clone':
-            prefix = 'self' if which == 'clone' else 'base'
+        if which in ('clone', 'generate'):
```
Comment (jyejare) [inline nailgun/entities.py]: Just add a docstring for api_path of generate(just like clone) in this 'path' method and we are good to merge this PR.

Comment (lhellebr) [reply nailgun/entities.py]: Fixed. Also, I think "clone" docstring was incorrect so I fixed that, too.

**Pull request merged**